pmdarima/utils/visualization.py

```diff
@@ -58,6 +59,49 @@ def _show_or_return(obj, show):
         return obj
 
 
+def decomposed_plot(decomposed_tuple, figure_kwargs=None, show=True):
+    """
+    Plot the decomposition of a time series.
+
+    This includes the 'x', 'trend', 'seasonal', and 'random' components.
+
+    Parameters
+    ----------
+    decomposed_tuple : tuple
+        Tuple of series that consist of data, trend, seasonal, and random.
+
+    figure_kwargs : dict, optional (default=None)
+        Optional dictionary of keyword arguments that are passed to figure.
+
+    show : bool, optional (default=True)
+        Whether to show the plot after it's been created. If not, will return
+        the plot as an Axis object instead.
+
+    Notes
+    -----
+    This method will only show the plot if ``show=True`` (which is the default
+    behavior). To simply get the axis back (say, to add to another canvas),
+    use ``show=False``.
+
+    """
+
+    _err_for_no_mpl()
+
+    fig, axes = mpl.subplots(4, 1, sharex=True, **figure_kwargs)
+
+    y_labels = ['data', 'trend', 'seasonal', 'random']
+
+    for ax in axes.flat:
```

Comment:
This is a neat trick, didn't know you could do that.

Reply:
Yup! So it is technically a `numpy` array under the hood so you can do that.